### PR TITLE
Add `required-labels` input.

### LINKED ISSUE #69: Automerge only when has certain label
This action seems to be exactly what I needed! 1 question: Instead of not auto merging it when it has a certain label, I would like to only auto merge it when it has a certain label. Does this action support that?

### PR DESCRIPTION
**Pull request by rgmvisser**
Adding a feature where the user can specify new input `required-labels`. If specified it will check that that the PR has this label applied and otherwise not auto-merge.

Closes #69

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -27,6 +27,10 @@ inputs:
     required: false
     description: >
       When any of the labels in this comma-separated list is applied to a pull request, it will not be merged automatically.
+  required-labels:
+    required: false
+    description: >
+      Comma-separated list of labels that is required to be applied ot the pull request before merging automatically.
```
Comment (reitermarkus) [inline action.yml:33]:
```suggestion
      Comma-separated list of labels that are required to be applied to a pull request for it to be merged automatically.
```

**Review by reitermarkus — CHANGES_REQUESTED** [review]

File: src/automerge-action.ts [hunk 2]
```diff
@@ -116,6 +116,17 @@ export class AutomergeAction {
       return false
     }
 
+    if (this.input.requiredLabels.length > 0) {
```
Comment (reitermarkus) [inline src/automerge-action.ts:119]: I think this is unneeded, since the loop won't run anyways if it's empty.

Comment (rgmvisser) [reply src/automerge-action.ts:119]: You're right, this was more for clarity, but will remove it.

**Review by reitermarkus — APPROVED** [review]

**Comment by reitermarkus** [comment]
Thanks, @rgmvisser!

**Comment by reitermarkus** [comment]
@rgmvisser, I guess the action should now also support running when a pull request is labelled with a required label?

**Comment by rgmvisser** [comment]
> @rgmvisser, I guess the action should now also support running when a pull request is labelled with a required label? Mind also adding support for this?

I am not too familiar with actions, so not sure how to do this. Do you have an example of this? Happy to add it then

**Comment by reitermarkus** [comment]
The `automerge.yml` would have this trigger:

```yml
  pull_request_target:
    types:
      - labeled
```

Additionally there should be a check in `main.ts` (or inside the `handlePullRequestTarget` function) so it only runs on that trigger when a required label is applied. I guess the same should also apply when a “do-not-merge” is removed.

**Pull request merged**